File: solar_doomsday/solarDoomsday.java
import java.lang.Math.sqrt;

// Not running correctly (Python code submitted)
public class Solution {
    public static int[] solution(int area) {
        ArrayList<int> ans = new ArrayList<int>();
        while (area > 0) {
            int x = (int)(sqrt((double)(area)));
            x *= x;
            area -= x;
            ans.add(x);
        }
        return ans.toArray();
    }
}

File: solar_doomsday/solar_doomsday.py
from math import sqrt, floor


def solution(area):
    ans = []
    while area > 0:
        # Note that Python 2.7 doesn't have math.isqrt
        nearestSquare = int(floor(sqrt(area)))**2
        area -= nearestSquare
        ans.append(nearestSquare)
    return ans
